x-pack/plugins/task_manager/server/task_scheduling.ts

```diff
@@ -0,0 +1,176 @@
+    options?: Record<string, unknown>
+  ): Promise<ConcreteTaskInstance> {
+    const { taskInstance: modifiedTask } = await this.middleware.beforeSave({
+      ...options,
+      taskInstance: ensureDeprecatedFieldsAreCorrected(taskInstance, this.logger),
+    });
+    return await this.store.schedule(modifiedTask);
+  }
+
+  /**
+   * Run  task.
+   *
+   * @param taskId - The task being scheduled.
+   * @returns {Promise<ConcreteTaskInstance>}
+   */
+  public async runNow(taskId: string): Promise<RunNowResult> {
+    return new Promise(async (resolve, reject) => {
+      this.awaitTaskRunResult(taskId).then(resolve).catch(reject);
+      this.taskPollingLifecycle.attemptToRun(taskId);
+    });
+  }
+
+  /**
+   * Schedules a task with an Id
+   *
+   * @param task - The task being scheduled.
+   * @returns {Promise<TaskInstanceWithId>}
+   */
+  public async ensureScheduled(
+    taskInstance: TaskInstanceWithId,
+    options?: Record<string, unknown>
+  ): Promise<TaskInstanceWithId> {
+    try {
+      return await this.schedule(taskInstance, options);
+    } catch (err) {
+      if (err.statusCode === VERSION_CONFLICT_STATUS) {
+        return taskInstance;
+      }
+      throw err;
+    }
+  }
+
+  private async awaitTaskRunResult(taskId: string): Promise<RunNowResult> {
+    return new Promise((resolve, reject) => {
+      const subscription = this.taskPollingLifecycle.events
+        // listen for all events related to the current task
+        .pipe(filter(({ id }: TaskLifecycleEvent) => id === taskId))
+        .subscribe((taskEvent: TaskLifecycleEvent) => {
+          if (isTaskClaimEvent(taskEvent)) {
+            mapErr(async (error: Option<ConcreteTaskInstance>) => {
+              // reject if any error event takes place for the requested task
+              subscription.unsubscribe();
+              return reject(
+                map(
+                  await pipe(
+                    error,
+                    mapOptional(async (taskReturnedBySweep) => asOk(taskReturnedBySweep.status)),
+                    getOrElse(() =>
+                      // if the error happened in the Claim phase - we try to provide better insight
+                      // into why we failed to claim by getting the task's current lifecycle status
+                      promiseResult<TaskLifecycle, Error>(this.store.getLifecycle(taskId))
+                    )
+                  ),
+                  (taskLifecycleStatus: TaskLifecycle) => {
+                    if (taskLifecycleStatus === TaskLifecycleResult.NotFound) {
+                      return new Error(`Failed to run task "${taskId}" as it does not exist`);
+                    } else if (
+                      taskLifecycleStatus === TaskStatus.Running ||
+                      taskLifecycleStatus === TaskStatus.Claiming
+                    ) {
+                      return new Error(`Failed to run task "${taskId}" as it is currently running`);
```

Comment:
@mikecote and I actually just discussed this today - we do think this is needed as we'll likely want to add a "test alert" in the UI in the future and this would be displayed to the user.

In anycase -  this wasn't added in this PR, just moved.
Happy to revisit in a different issue if you'd like 🤷